src/core/logging.js

```diff
@@ -73,3 +64,41 @@ function verifyLoggingCallbacks() {
 		}
 	}
 }
+
+function emit( type, data ) {
+	var i, len, callbacks;
+
+	// Validate
+	if ( QUnit.objectType( type ) !== "string" ) {
+		throw new Error( "Emitting QUnit events requires an event type" );
+	}
+
+	callbacks = listeners[ type ];
+	if ( callbacks ) {
+		for ( i = 0, len = callbacks.length; i < len; i++ ) {
```

Comment:
That's interesting, didn't see it as an anti-pattern anywhere yet. I'll change it anyway.